### ISSUE TITLE
Azure Inventory Plugin: Feature flag to import public IP addresses of scale sets

### ISSUE THREAD

**Issue by fivetran-artemanikieiev** [ansible/ansible_collections_azure#1775]
##### SUMMARY
After importing Virtual Machine Scale sets into AWX inventory using this configuration:
```yaml
include_vmss_resource_groups:
  - '*'
```
  
 I see that Ansible imported private ip addresses of the instances. Each instance has public IP address allocated from public ip address prefix. This configuration is supported in Azure (https://learn.microsoft.com/en-us/azure/virtual-network/ip-services/public-ip-address-prefix). Is there any specific configuration which I need to set up?

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
Azure Inventory

##### ADDITIONAL INFORMATION

```yaml
include_vmss_resource_groups:
  - '*'
```

**Comment by Fred-sun** [comment]
@fivetran-artemanikieiev Thank you for submitting the problem. Do you want to get the information about `Public IP address prefix` when you get the VM information? Thank you!

**Comment by Fred-sun** [comment]
@fivetran-artemanikieiev Now we can return the public IP of VMSS, uniform VMSS instance VM does not have a public IP, return empty, the return information is as follows. Thank you!
```e.g.
Flexible VMSS:
                ......
                "public_dns_hostnames": [],
                "public_ip_address": [
                    {
                        "id": "/subscriptions/xxxxxxxxx/resourceGroups/v-xisuRG02/providers/Microsoft.Network/publicIPAddresses/publicIp-vnet-eastus-nic01-2bacd76f",
                        "ipv4_address": "13.82.109.204",
                        "name": "publicIp-vnet-eastus-nic01-2bacd76f"
                    }
                ],
                "public_ipv4_address": [
                    "13.82.109.204"
                ],
                ......
```
```e.g.
Uniform VMSS:
                ......
                "provisioning_state": "succeeded",
                "public_dns_hostnames": [],
                "public_ip_address": [],
                "public_ipv4_address": [],.
                ......

**Comment by fivetran-artemanikieiev** [comment]
@Fred-sun which plugin version and configuration options will be required for this? Thank you very much for the responce! To clarify I want to get public IP address assigned to VMSS instance. Public IP address was assigned from Public IP Prefix.

For example:
We have `20.20.20.248/29` prefix
and VMSS instance has `20.20.20.251` Public IP address assigned from that prefix. In Azure UI I am able to see it has public IP, but my current configuration imports only private IP address. In my case Orchestration Mode is Uniform

![Image](https://github.com/user-attachments/assets/921a4ca5-f3bc-4d21-8d44-2143e968015b)

**Comment by Fred-sun** [comment]
@fivetran-artemanikieiev It is found that PublicIP Address is configured only for Flexible mode VMSS instances, but not for Uniform mode VMSS instances. Thank you!

So there's no problem with Unform mode VMSS instances not having PublicIP addresses。

**Comment by fivetran-artemanikieiev** [comment]
@Fred-sun yes. So the problem is Uniform mode VMSS instances have public IP when it is derived from public IP Prefix, as evident on screenshot and awx inventory plugin does not export their public IP address

**Comment by Fred-sun** [comment]
@fivetran-artemanikieiev Because the network card management method (instance-level Public IP) of Uniform VMSS Instance is different from that of Flexible VMSS. I added obtaining the network card information of the Unform VMSS Instance in #2055. Could you update the changes in #2055 to verify if your issue has been resolved? Thank you very much!

**Comment by fivetran-artemanikieiev** [comment]
@Fred-sun I think we can close the issue. I see in our ansible instance we are able to resolve external IP addresses of Azure VMs in VMSS. Thank you for fix!
